tests/yam/agama/patch_agama.pm

```diff
@@ -0,0 +1,35 @@
+## Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: This module use yupdate patch the Agama on Live Medium
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+
+sub run {
+    assert_screen('agama_product_selection', 120);
+
+    select_serial_terminal;
```

Comment:
serial console will give more screen indeed but it will not change current tty console, if you switch to root console then you should do select_console(installation) back, otherwise the next module agama_installation will failed, that's why i select serial console finally.

Reply:
I'd prefer to switch back, to have too many screenshot seems to be a problem for developers trying to figure out what the test does. For one extra line of code one can try to do something with the feedback we get.

Reply:
If you want select_console 'installation' can take effect you need more setting such as VIDEOMODE to text.
But following failure will happen if you set VIDEOMODE=text.
https://openqa.opensuse.org/tests/3346366#step/bootloader_start/14

Additional info:
lib/susedistribution.pm:490:        $self->add_console('installation', 'tty-console', {tty => check_var('VIDEOMODE', 'text') ? 1 : 7});

Reply:
So i prefer still using select_serial_terminal, what do you think?

Reply:
never checked that code, didn't know there was those difficulties, could you please create a ticket as well for handling this difficulty.

Reply:
Sorry my misleading, the VIDEOMODE should not be setted since agama's UI tty is on 7.  
So i rebase the code again.  We do not need new ticket for investigation this.